### PR TITLE
fixed hashmap collection with StaticArray<u8> usage and Result types

### PR DESCRIPTION
**Pull request by BatiGencho**

### CODE REVIEW

File: assembly/collections/persistentMap.ts [hunk 1]
```diff
@@ -206,29 +220,36 @@ export class PersistentMap<K, V> {
...
+    } else if (isArrayLike<V>()) {
+      this._increaseSize(key);
+      // @ts-ignore
+      Storage.set(this._key(key), unwrapStaticArray(value));
     } else {
       this._increaseSize(key);
       // @ts-ignore
       Storage.set(this._key(key), value.toString());
     }
+    return new Result(this.size(), null);
```
Comment (Thykof) [inline assembly/collections/persistentMap.ts:253]: I removed some @ts-ignore with `as` and moved _increaseSize before `if`

```ts

  /**
   * @example
   * ```ts
   * let map = new PersistentMap<string, string>("m")
   *
   * map.set("hello", "world")
   * ```
   *
   * Sets the new value for the given key.
   * @param key - Key of the element.
   * @param value - The new value of the element.
   */
  set(key: K, value: V): Result<usize> {
    // check for map size wont overflow
    if (this._size >= Usize.MAX_VALUE) {
      return new Result(this.size(), 'map size overflow');
    }

    this._increaseSize(key);

    if (isString<V>()) {
      Storage.set(this._key(key), stringToBytes(value as string));
    } else if (isInteger<V>()) {
      Storage.set(this._key(key), i64ToBytes(value as i64));
    } else if (isFloat<V>()) {
      Storage.set(this._key(key), f64ToBytes(value as f64));
    } else if (isBoolean<V>()) {
      Storage.set(this._key(key), boolToByte(value as boolean));
    } else if (isArrayLike<V>()) {
      Storage.set(this._key(key), unwrapStaticArray(value as Uint8Array));
    } else {
      // @ts-ignore
      Storage.set(this._key(key), value.toString());
    }

    return new Result(this.size());
  }
```

**Review by Thykof — CHANGES_REQUESTED** [review]

**Review by peterjah — APPROVED** [review]

File: assembly/collections/persistentMap.ts (hunk 1 above)
Comment (BatiGencho) [reply assembly/collections/persistentMap.ts]: done, thanks

File: assembly/collections/persistentMap.ts [hunk 2]
```diff
@@ -188,11 +201,12 @@ export class PersistentMap<K, V> {
    * @param key - Key of the element.
    * @returns Value for the given key or the default value.
    */
-  getSome(key: K): V {
-    assert(this.contains(key), 'key not found');
+  getSome(key: K): Result<V> {
+    if (!this.contains(key)) {
+      return new Result(null, 'key not found');
```
[the remark this answers is not in the document]
Comment (BatiGencho) [reply assembly/collections/persistentMap.ts]: done, thanks

File: assembly/collections/persistentMap.ts [hunk 3]
```diff
@@ -188,11 +201,12 @@ export class PersistentMap<K, V> {
...
+    }
     const res = this.get(key, null);
-    assert(res, 'bad result');
-    return <V>res;
+    return new Result(<V>res, null);
```
[the remark this answers is not in the document]
Comment (BatiGencho) [reply assembly/collections/persistentMap.ts]: great, thanks

**Review by Thykof — APPROVED** [review]
we only do `npm update` package by package to not risk to break something

**Pull request merged**